src/Chainweb/BlockHeader/Validation.hs

```diff
@@ -403,7 +403,9 @@ prop_block_current :: Time Micros -> BlockHeader -> Bool
 prop_block_current t b = BlockCreationTime t >= _blockCreationTime b
 
 prop_block_featureFlags :: BlockHeader -> Bool
-prop_block_featureFlags b = _blockFlags b == mkFeatureFlags
+prop_block_featureFlags b
+    | skipFeatureFlagValidationGuard b = _blockFlags b == mkFeatureFlags
```

Comment:
This function (and perhaps this whole `prop_XXX`) is extremely confusing -- looking at the old code, one would think that `prop_block_featureFlags` was doing validation, after all it is testing that flags == 0. But instead, it's a `not` in `validateInstrinsic`, which is presumably the bug and why we weren't validating. In general, it's confusing to phrase rules in the negative, which is also why the predicate `skipFeature...` is also undesirable: we now have an even more confusing double-negative. Suggest instead:
- this property should be positive: it returns `True` if the flags are set _properly_, such that an error is `False`, updating `validateIntrinsic` accordingly.
- `skipFeatureFlagValidationGuard` should be positive: `enableFeatureFlagValidationGuard`.

Going forward please endeavor to follow this as it is widely agreed that reasoning in the negative is confusing. At a minimum, if something is in the negative the name MUST indicate it, ie `prop_invalid_feature_flags`. But I vastly prefer positive reasoning.